src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -650,6 +703,15 @@ def engine_forward(
             callback = context.get("callback")
             stop = context.get("stop")
 
+            max_new_tokens = context.get("max_new_tokens")
+            if max_new_tokens:
+                max_tokens = max_new_tokens + len(generated_tokens)
+            else:
+                max_tokens = context.get("max_tokens", 0)
+                max_tokens = (
+                    max_tokens if max_tokens > 0 else (100 * self.sequence_length)
+                )
```

Comment:
looks correct to me - 
max new tokens only counts generated ones
max tokens counts the prompt tokens as well